pkg/backend/openshiftcluster.go

```diff
@@ -33,8 +33,7 @@ func (ocb *openShiftClusterBackend) try(ctx context.Context) (bool, error) {
 
 	log := ocb.baseLog.WithField("resource", doc.OpenShiftCluster.ID)
 	if doc.Dequeues > maxDequeueCount {
-		log.Errorf("dequeued %d times, failing", doc.Dequeues)
-		return true, ocb.endLease(ctx, nil, doc, api.ProvisioningStateFailed)
+		return true, ocb.endLease(ctx, nil, doc, api.ProvisioningStateFailed, fmt.Errorf("dequeued %d times, failing", doc.Dequeues))
```

Comment:
if you want to do:
```
err := fmt.Errorf("dequeued %d times, failing", doc.Dequeues)
log.Error(err)
return true, ocb.endLease(ctx, nil, doc, api.ProvisioningStateFailed, err)
```
that's OK

Reply:
The error will be logged by `endLease`. This way we simplify logging of the errors when the lease is ended.